conductr_cli/conduct_acls.py

```diff
@@ -0,0 +1,157 @@
+from conductr_cli import bundle_utils, conduct_url, validation, screen_utils
+import json
+import logging
+import re
+import requests
+from conductr_cli.http import DEFAULT_HTTP_TIMEOUT
+
+
+SUPPORTED_PROTOCOL_FAMILIES = ['http', 'tcp']
+ALL_HTTP_METHOD = '*'
+HTTP_ACL_PATH = 'path'
+HTTP_ACL_PATH_BEG = 'path_beg'
+HTTP_ACL_PATH_REGEX = 'path_regex'
+
+
+@validation.handle_connection_error
+@validation.handle_http_error
+def acls(args):
+    """`conduct acls` command"""
+
+    log = logging.getLogger(__name__)
+    url = conduct_url.url('bundles', args)
+    response = requests.get(url, timeout=DEFAULT_HTTP_TIMEOUT)
+    validation.raise_for_status_inc_3xx(response)
+
+    if log.is_verbose_enabled():
+        log.verbose(validation.pretty_json(response.text))
+
+    all_acls = [
+        {
+            'acl': acl,
+            'bundle_id': bundle['bundleId'] if args.long_ids else bundle_utils.short_id(bundle['bundleId']),
+            'bundle_name': bundle['attributes']['bundleName'],
+            'status': 'Running' if execution['isStarted'] else 'Starting'
+        }
+        for bundle in json.loads(response.text)
+        for execution in bundle['bundleExecutions']
+        for endpoint_name, endpoint in bundle['bundleConfig']['endpoints'].items() if 'acls' in endpoint
+        for acl in endpoint['acls']
+    ]
+
+    if args.protocol_family == 'http':
+        http_acls = [acl for acl in all_acls if 'http' in acl['acl']]
+        display_http_acls(log, http_acls)
+
+    elif args.protocol_family == 'tcp':
+        tcp_acls = [acl for acl in all_acls if 'tcp' in acl['acl']]
+        display_tcp_acls(log, tcp_acls)
+
+    return True
+
+
+def display_tcp_acls(log, tcp_acls):
+    def tcp_port_value(line):
+        return int(line['tcp_port'])
+
+    data = [
+        {
+            'tcp_port': 'TCP/PORT',
+            'bundle_id': 'BUNDLE ID',
+            'bundle_name': 'BUNDLE NAME',
+            'status': 'STATUS'
+        }
+    ] + sorted([
+        {
+            'tcp_port': tcp_port,
+            'bundle_id': tcp_acl['bundle_id'],
+            'bundle_name': tcp_acl['bundle_name'],
+            'status': tcp_acl['status']
+        }
+        for tcp_acl in tcp_acls
+        for tcp_port in tcp_acl['acl']['tcp']['requests']
+    ], key=tcp_port_value)
```

Comment:
Ordering of the TCP based ACL will be displayed according to what ConductR is doing, i.e. sorted by port number.